packages/storybook/stories/va-file-input.stories.jsx

```diff
@@ -0,0 +1,135 @@
+/* eslint-disable react/prop-types */
+import React, { useState, useEffect, useCallback, useRef } from 'react';
+import { getWebComponentDocs, propStructure, StoryDocs } from './wc-helpers';
+
+const fileInputDocs = getWebComponentDocs('va-file-input');
+
+export default {
+  title: 'Components/va-file-input',
+  parameters: {
+    componentSubtitle: `File input web component`,
+    docs: {
+      page: () => <StoryDocs data={fileInputDocs} />,
+    },
+  },
+};
+
+const defaultArgs = {
+  'label': 'This is the file upload label',
+  'name': 'my-file-input',
+  'buttontext': 'Upload a document or file',
+  'accept': 'image/*',
+  'required': false,
+  'multiple': false,
+  'error': null,
+  'enable-analytics': false,
+};
+
+const handleChangeEvent = (event) => {
+  alert(`Uploaded event received: ${event?.detail?.files[0]?.name}`)
+};
+
+const Template = ({
+  label,
+  name,
+  buttontext,
+  accept,
+  error,
+  required,
+  multiple,
+  'enable-analytics': enableAnalytics,
+}) => {
+  const ref = useRef(null);
+
+  useEffect( () => {
+    const element = ref.current;
+    element.addEventListener('vaChange', handleChangeEvent);
+
+    return () => {
+      element.removeEventListener('vaChange', handleChangeEvent);
+    };
+  }, [])
+
+  return (
+    <va-file-input
+      ref={ref}
+      label={label}
+      name={name}
+      buttontext={buttontext}
+      accept={accept}
+      required={required}
+      multiple={multiple}
+      error={error}
+      enable-analytics={enableAnalytics}
+    />
+  );
```

Comment:
I like what you're doing here, but I think it makes it slightly more difficult for VFS teams to use. We can use the [bindings](https://stenciljs.com/docs/framework-bindings) created by Stencil using `yarn build-bindings` for this component here instead. A benefit of using bindings is VFS engineers using this component in React applications can use `onVaChange` as a prop and pass `handleChangeEvent` to it without having to add and remove event listeners.

Reply:
Here's an [example](https://github.com/department-of-veterans-affairs/component-library/blob/main/packages/storybook/stories/va-modal.stories.jsx#L27-L74) using bindings in Storybook.

Reply:
@andresriveratoro Thank you for the direction here! I'll check this out.